### PR TITLE
Stat Power ups + Power up framework

### LINKED ISSUE #12: Implement power up reward mechanism
There needs to be a method or interface for the player to choose their reward, triggered at the end of rooms or waves.
Currently the idea is that of Vampire Survivors, where the player is presented with a UI menu with the choices.

### PR DESCRIPTION
**Pull request by zhuandy42**
Changes
* PowerUp framework implemented
* Stat PowerUps implemented except for Reload Speed

Summary
powerups are sorted into classes, statpowerups are applied directly and are not stored. the other classes will be stored in an array in powerupmanager that we will look through when we need to apply the effects of the powerup.

to trigger the entire powerup framework, call presentPowerUp() from PowerUpManager.

closes #12 and #7 , will be closing previous PR #37

### CODE REVIEW

File: Assets/Scripts/Weapons/Weapon.cs [hunk 1]
```diff
@@ -13,17 +13,16 @@ public class Weapon : MonoBehaviour
         { "repeater", false }
     };
 
-    AbstractAttackBehaviour _attackBehaviour;
-    float _fireRate; // shots per second
-    int _magazineSize;
+    public AbstractAttackBehaviour _attackBehaviour;
+    public float _fireRate; // shots per second
+    public int _magazineSize;
```
Comment (zhuandy42) [inline Assets/Scripts/Weapons/Weapon.cs:18]: let me know if we shouldn't expose these fields, I changed them to public to get access to apply the powerups

File: Assets/Scripts/PowerUps/AbstractPowerUp.cs [hunk 2]
```diff
@@ -0,0 +1,10 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public abstract class AbstractPowerUp
+{
+    public float modifier;
+    public abstract Vector3[] applyPowerUp(Vector3[] direction);
```
Comment (Sirkular) [inline Assets/Scripts/PowerUps/AbstractPowerUp.cs:8]: Only BulletPowerUps should have this signature as their applyPowerUp. If this limits implementation, perhaps we should remove this base class.

File: Assets/Scripts/SpeedManager.cs [hunk 3]
```diff
@@ -20,9 +21,33 @@ public class SpeedManager : MonoBehaviour
     }
 
     static void updateGameObjectSpeed(){
-        playerMovementScaling = coreSpeed;
+        playerMovementScaling = coreSpeed + adrenalin();
         bulletSpeedScaling = coreSpeed;
         enemySpawnScaling = 1 / coreSpeed;
     }
 
+    static float adrenalin(){
```
Comment (Sirkular) [inline Assets/Scripts/SpeedManager.cs:29]: Interesting!

**Review by Sirkular — APPROVED** [review]
There's a bit of spaghetti with AbstractPowerUp, but it's fine.

**Pull request merged**